library.py

```diff
@@ -0,0 +1,110 @@
+from pydantic import BaseModel
+from abc import ABC, abstractmethod
+
+class Publication(ABC):
+    @abstractmethod
+    def get_title(self):
+        pass
+
+    @abstractmethod
+    def get_autor(self):
+        pass
+
+    @abstractmethod
+    def get_year(self):
+        pass
+class Bookmodel(BaseModel):
+    title: str
+    year: int
+    autor: str
+
+
+class Book(Publication):
+    def __init__(self,title,year,autor):
+        self.__title=title
+        self.__year=year
+        self.__autor=autor
+
+    def __str__(self):
+        return Book(f'{self.title} by {self.autor} from {self.year}')
+
+class Library:
+    def __init__(self):
+        self.books=[]
+
+    def __iter__(self):
+        return iter(self.books)
+    def generator_books(self,autor: str):
+        for book in self.books:
+            if book.autor == autor:
+                yield book
+
+    def log_add_book(func):
+        def wrapper(*args, **kwargs):
+            result = func(*args, **kwargs)
+            print("New book added to the library!")
+            return result
+        return wrapper
+
+    @log_add_book
+    def add_book(self, book: Book):
+        self.books.append(book)
+
+    def check_del_book(func):
+        def wrapper(self,book):
+            if book in self.books:
+                print('this book exist in library')
+            return func(self,book)
```

Comment:
```suggestion
            return func(self, book)
```
space between arguments